src/sentry/options/manager.py

```diff
@@ -178,8 +197,10 @@ def register(self, key, default='', type=None, flags=DEFAULT_FLAGS,
         # value instead that matches the type expected, rather than relying on None.
         if type is NoneType:
             raise TypeError('Options must not be NoneType')
-        if not isinstance(default, type):
+        if default is not None and not self._value_is_of_type(type, default):
             raise TypeError('got %r, expected %r' % (_type(default), type))
+        if default is None:
+            default = type()
```

Comment:
I guess this is a bit nicer.

Since we're doing this now, you can remove the `default={}` on the existing options.